packages/material-ui/src/internal/SelectableGroup/SelectableGroup.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as utils from './utils';
+import SelectableGroupContext from './SelectableGroupContext';
+import is from 'object.is';
+
+/**
+ * @ignore - internal component.
+ */
+function SelectableGroup(props) {
+  const [selected, setSelected] = React.useState(null);
+  const { additional, children, exclusive, value: valueProp } = props;
+  const { current: isControlled } = React.useRef(valueProp != null);
+
+  const createReducer = handler => {
+    return (value, event) => {
+      const newSelected = handler(selected, value, exclusive);
+
+      setSelected(newSelected);
+
+      // Only call onChange if it exists and state has changed (Object.is mimics React)
+      if (props.onChange && !is(newSelected, selected)) {
+        props.onChange(event, newSelected);
+      }
+    };
+  };
+
+  if (isControlled && valueProp !== selected) {
+    setSelected(valueProp);
+  }
+
+  const deselect = createReducer(utils.deselect);
+  const select = createReducer(utils.select);
+  const toggle = createReducer(utils.toggle);
+  const isValueSelected = value => utils.hasValue(selected, value);
+
+  return (
+    <SelectableGroupContext.Provider
+      value={{
+        additional,
+        deselect,
+        isValueSelected,
+        select,
+        toggle,
+      }}
+    >
+      {children}
+    </SelectableGroupContext.Provider>
+  );
+}
+
+SelectableGroup.propTypes = {
+  /**
+   * Additional props to pass down.
+   */
+  additional: PropTypes.object,
```

Comment:
Any ideas for a better name? `additional` does sound like it affects behavior. I would rather let the components defined their own context in those cases. This will likely be an object at which point future memoization optimizations might be tricky.

Reply:
Currently, it's only used to pass down the name prop. I'm fine with just replacing additional with name.